root/static/scripts/common/i18n/expand2text.js

```diff
@@ -0,0 +1,70 @@
+/*
+ * @flow
+ * Copyright (C) 2018 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+import * as React from 'react';
+
+import expand, {
+  createCondSubstParser,
+  createTextContentParser,
+  parseContinuousString,
+  parseStringVarSubst,
+} from './expand2';
+
+const textContent = /^[^{}]+/;
+const condSubstThenTextContent = /^[^{}|]+/;
+const parseRootTextContent = createTextContentParser(textContent);
+
+const parseCondSubst = createCondSubstParser<string, StrOrNum>(
+  args => parseContinuousString<StrOrNum>(condSubstThenParsers, args),
+  args => parseContinuousString<StrOrNum>(condSubstElseParsers, args),
+);
+
+const parseCondSubstThenTextContent = createTextContentParser(
+  condSubstThenTextContent,
+);
+
+const condSubstThenParsers = [
+  parseCondSubstThenTextContent,
+  parseStringVarSubst,
+  parseCondSubst,
+];
+
+const condSubstElseParsers = [
+  parseRootTextContent,
+  parseStringVarSubst,
+  parseCondSubst,
+];
+
+const rootParsers = [
+  parseRootTextContent,
+  parseStringVarSubst,
+  parseCondSubst,
+];
+
+function parseRoot(args) {
+  return parseContinuousString<StrOrNum>(rootParsers, args);
+}
+
+/*
+ * Like expand2react, but can only interpolate and produce plain text
+ * values. To be more specific: HTML isn't parsed, link interpolation
+ * syntax isn't supported, and only strings and numbers can be used
+ * as var args. The result is always a string.
+ *
+ * This is useful in cases where strings are /required/, such as HTML
+ * title or aria attributes and select option values. It can also be
+ * used where plain text is expected (if not required), simply for the
+ * stricter types.
+ */
+export default function expand2text(
+  source: string,
+  args?: ?{+[string]: StrOrNum},
```

Comment:
Shouldn’t `args` object be mandatory?

Reply:
Yeah, for `expand2text` it should. (Not `expand2react`, since there are places like [here](https://github.com/metabrainz/musicbrainz-server/blob/eb98965/root/edit/components/EditSidebar.js#L61) where it's only used to convert HTML.)

One issue you've made me notice: `expand2text` still decodes HTML entities. So `expand2text('&lt;')` would become `'<'` (meaning it isn't entirely a no-op if no args are provided). I think this should perhaps be changed, since it doesn't handle any other HTML. Though it will require translators to be aware that they can probably only use HTML entities if the original string uses them. (I'm not sure why they would need to otherwise.)